src/encoded/static/components/static-pages/placeholders/PageCarousel.js

```diff
@@ -0,0 +1,211 @@
+     * @param {Object} slide    Carousel slide to be displayed
+     */
+    renderSlide(slide) {
+        const { img, description, title, badge, badgeBgColor, link } = slide;
+        const { slideHeight, adjustImageHeight, textPosition } = this.props;
+
+        //todo: move css rules to _static-pages.scss after usability tests
+        // let titleContainerStyle = null;
+        // let badgeOuterStyle = null;
+        // if (textPosition === 'bottom') {
+        //     titleContainerStyle = { left: 0, right: 0, top: 'unset', bottom: 0 };
+        //     badgeOuterStyle = { left: '10px', right: 'unset' };
+        // } else if (textPosition === 'top') {
+        //     titleContainerStyle = { left: 0, right: 0, top: 0, bottom: 'unset' };
+        //     badgeOuterStyle = { left: 'unset', right: '10px' };
+        // } else if (textPosition === 'left') {
+        //     titleContainerStyle = { left: 0, right: 'unset', top: 0, bottom: 0, width: '35%' };
+        //     badgeOuterStyle = { left: 'unset', right: '10px' };
+        // } else if (textPosition === 'right') {
+        //     titleContainerStyle = { left: 'unset', right: 0, top: 0, bottom: 0, width: '35%' };
+        //     badgeOuterStyle = { left: '10px', right: 'unset' };
+        // }
+        const content = ((title || description) ?
+            (
+                <div className="title-container">
+                    {title ? <h4 className="mt-0">{title}</h4> : null}
+                    {description ? <p>{description}</p> : null}
+                </div>
+            ) : null);
+
+        const containerStyle = { height: slideHeight };
+        const badgeStyle = badgeBgColor ? { backgroundColor: badgeBgColor } : null;
+        const innerFrame = (
+            <div style={containerStyle}>
+                <div className="inner-container" style={containerStyle}>
+                    <div className="bg-image" style={img ? { 'backgroundImage': 'url(' + img + ')' } : null} />
+                    {content}
+                </div>
+                <div className="inner-body">
+                    {badge ?
+                        <div className="inner-body">
+                            <div className="slide-label" style={badgeStyle}>{badge}</div>
+                        </div>
+                        : null}
+                </div>
+            </div>
+        );
+
+        if (link) {
+            const isExternalLink = link.slice(0, 4) === 'http';
+            return (
+                <a className="homepage-carousel-slide is-link" href={link} target={isExternalLink ? '_blank' : null} rel={isExternalLink ? 'noopener noreferrer' : null}>
+                    {innerFrame}
+                </a>
+            );
+        }
+
+        return <div className="homepage-carousel-slide">{innerFrame}</div>;
+    }
+
+    render() {
+        const { slides, slidesToShow, navControlPosition, slideHeight, adjustImageHeight, showSlideCount, textPosition, windowWidth } = this.props;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['textPosition' is missing in props validation (react/prop-types)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4476879)